.github/workflows/docker-unified.yml

```diff
@@ -37,8 +37,29 @@ env:
   DATAHUB_INGESTION_IMAGE: "acryldata/datahub-ingestion"
 
 jobs:
+  check_lint:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out the repo
+        uses: hsheth2/sane-checkout-action@v1
+      - uses: actions/setup-python@v4
+        with:
+          python-version: "3.10"
+          cache: "pip"
+      - name: Set up JDK 17
+        uses: actions/setup-java@v3
+        with:
+          distribution: "zulu"
+          java-version: 17
+      - name: Ensure packages are correct
+        run: |
+          python ./.github/scripts/check_python_package.py
+      - name: Run lint on smoke test
+        run: |
+          ./gradlew :smoke-test:lint
   setup:
     runs-on: ubuntu-latest
+    needs: check_lint
```

Comment:
Isn’t this still a blocking step? The needs annotation should be on the smoke test job

Reply:
The reason for moving this was because David mentioned it was not a clean approach to add this to setup. This is technically not setup.

It is still blocking. As it is intended that if smoke test lint is not working we don't want to spend time building docker images at all.